outer.c

```diff
@@ -20,14 +20,27 @@
 #include <unistd.h>
 
 #include "capable.h"
+#include "compat.h"
 #include "enter.h"
 #include "outer.h"
 #include "path.h"
+#include "preprocessor.h"
 #include "sig.h"
 #include "userns.h"
 #include "util.h"
 #include "fd.h"
 
+# define REBIND_FDS_AND_CLOSE(Start, ...) \
+	do { \
+		int __fd_last = (Start); \
+		APPLY(_REBIND_FD, __VA_ARGS__) \
+		if (bst_close_range(__fd_last, ~0U, 0) == -1) { \
+			err(1, "close_range"); \
+		} \
+	} while (0)
```

Comment:
Is this because you want to scope __fd_last? Do you need the `do/while` ? Can't you just do the brackets without an operator?

There's versions of C were this works - A professor used to call this "the scope operator" but I haven't seen tat term used in any other source
```
{
    blah blah blah
}
```